extensions/Push/lib/Connector/Webhook.pm

```diff
@@ -94,17 +96,35 @@ sub send {
     $payload->{webhook_name} = $webhook->name;
     $payload->{webhook_id}   = $webhook->id;
 
-    my $bug_data   = $self->_get_bug_data($payload);
-    my $is_private = $bug_data->{is_private};
-    if ($is_private){
-      delete @{$payload}{bug};
-      if($payload->{event}->{action} eq 'modify'){
+    my $bug_data       = $self->_get_bug_data($payload);
+    my $bug_is_private = $bug_data->{is_private};
+    my $target         = $payload->{event}->{target};
+
+    my $target_is_private = $payload->{$target}->{is_private};
+    if ($target_is_private && ($target eq 'attachment' || $target eq 'comment')){
+      my $target_id = $payload->{$target}->{id};
+      delete @{$payload}{$target};
```

Comment:
For comments, if the bug is private or the comment is private, we need to remove the content for the comment. For example if we add a comment to a private bug, we need to make sure to remove `$payload->{comment}->{body}`.